src/check-for-ami.js

```diff
@@ -0,0 +1,45 @@
+let assert = require('assert');
+
+/**
+ * Check for the existence of an AMI that's executable by
+ * the provided ec2 object in the given region
+ */
+
+let debug = require('debug')('aws-provisioner:lib:check-for-ami');
+
+module.exports = async function (ec2, ami, region) {
+  let request = {
+    /* Not sure why this isn't working
+    ExecutableUsers: [
+      'self',
+      'all',
+    ],*/
+    ImageIds: [ami],
+  };
+
+  let result;
+  try {
+    // If we're using the multi-region wrapper, we handle things
+    // slightly differently
+    if (ec2.describeImages && ec2.describeImages.inRegion) {
+      result = await ec2.describeImages.inRegion(region, request);
```

Comment:
i wrote this patch knowing the drop would happen, but I'll remove the transitional code any way!